.github/workflows/scorecard.yml

```diff
@@ -0,0 +1,73 @@
+# This workflow uses actions that are not certified by GitHub. They are provided
+# by a third-party and are governed by separate terms of service, privacy
+# policy, and support documentation.
+
+name: Scorecard supply-chain security
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  # To guarantee Maintained check is occasionally updated. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained
+  schedule:
+    - cron: '15 20 * * 1'
+  push:
+    branches: [ "main" ]
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecard analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+      # Uncomment the permissions below if installing in a private repository.
+      # contents: read
+      # actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@0864cf19026789058feabb7e87baa5f140aac736 # v2.3.1
```

Comment:
Any specific reason for pinning to a specific commit sha instead of a version?

Reply:
The reason is that one of the recommendations from the OpenSSF is to use hash instead of tags, as it could lead to tags being modified or to operational errors such as using a tag like @2.x.x

GitHub has responded by creating immutable tags which are a bit new, and there is even a feature request to both scorecard and the scorecard action to use immutable tags as they are human readable. You can read a bit about it here.

https://github.com/ossf/scorecard/issues/4489

The creation of this scorecard.yml file was done by just following the steps in the installation section of this guide. (which is why it also came with comments)

https://github.com/ossf/scorecard-action